src/libponyc/pass/syntax.c

```diff
@@ -696,14 +696,19 @@ static ast_result_t syntax_consume(pass_opt_t* opt, ast_t* ast)
   {
     case TK_THIS:
     case TK_REFERENCE:
-    case TK_DOT:
       return AST_OK;
-
+    case TK_DOT: {
+        AST_GET_CHILDREN(term, left, right);
+        if (ast_id(left) != TK_CALL && ast_id(left) != TK_SEQ) 
+        {
+          return AST_OK;
+        }
```

Comment:
Small formatting fixes:

```suggestion
      AST_GET_CHILDREN(term, left, right);
      if (ast_id(left) != TK_CALL && ast_id(left) != TK_SEQ)
      {
        return AST_OK;
      }
```